### ISSUE TITLE
unindexed foreign keys should use order of the underlying unique index, not fkey definition

### ISSUE THREAD

**Issue by olirice** [supabase/splinter#87]
# Bug report
lints/0001_unindexed_foreign_keys.sql should enforce that multi-column indexes on the table declaring a foreign key have the same column order as the multi-column unique index being referenced. Column order differences between the two create less efficient query plans

```sql
create table foo(
	id int primary key,
	key1 int,
	key2 int,
	unique (key1, key2)
);

create table bar(
	id int primary key,
	key2 int,
	key1 int,
	foreign key (key2, key1) references foo(key2, key1)
);

set local enable_seqscan TO off;

-- Compare the following
-- create index on bar(key1, key2);
-- OR
-- create index on bar(key2, key1);

explain analyze
select *
from
	foo
	join bar
		on foo.key1 = bar.key1
		and foo.key2 = bar.key2;
```
<img width="802" alt="Screenshot 2024-06-20 at 2 44 03 PM" src="https://github.com/supabase/splinter/assets/12958657/eb97fdfd-f403-4939-b370-a1e53e84f061">
<img width="802" alt="Screenshot 2024-06-20 at 2 44 19 PM" src="https://github.com/supabase/splinter/assets/12958657/31da01a6-15c6-4767-93b2-6f73c1f91fde">

Currently, it enforces that the index matches the foreign key constraint definition, which is irrelevant for performance.
